library/core/src/iter/traits/iterator.rs

```diff
@@ -58,6 +58,10 @@ fn _assert_is_object_safe(_: &dyn Iterator<Item = ()>) {}
         note = "if you want to iterate between `start` until a value `end`, use the exclusive range \
               syntax `start..end` or the inclusive range syntax `start..=end`"
     ),
+    on(
+        any(_Self = "f32", _Self = "f64"),
```

Comment:
This only triggers on types which have been resolved to `f32` or `f64`. A floating point literal with no type annotation (as is the case in the issue this is fixing) has a `_Self` of `{float}`.

Reply:
Thanks for the tip.